contrib/docker/Dockerfile

```diff
@@ -0,0 +1,67 @@
+FROM alpine:3.22 AS builder
+
+RUN apk add --no-cache \
+    build-base \
+    cmake \
+    boost-dev \
+    libevent-dev \
+    sqlite-dev \
+    zeromq-dev \
+    coreutils \
+    binutils
+
+WORKDIR /opt/bitcoin
+
+COPY . .
+
+WORKDIR /opt/bitcoin/build
+
+RUN cmake .. \
+    -DCMAKE_INSTALL_PREFIX="/opt/bitcoin/build/" \
+    -DBUILD_DAEMON="ON" \
+    -DBUILD_CLI="ON" \
+    -DENABLE_WALLET="ON" \
+    -DWITH_ZMQ="ON" \
+    -DBUILD_GUI="OFF" \
+    -DBUILD_TESTS="OFF" \
+    -DBUILD_BENCH="OFF" \
+    -DBUILD_FUZZ_BINARY="OFF" \
+    -DBUILD_UTIL_CHAINSTATE="OFF" \
+    -DWITH_BDB="OFF" \
+    -DWITH_USDT="OFF" \
+    -DWITH_MULTIPROCESS="OFF" \
+    -DINSTALL_MAN="OFF" \
+    -DWITH_CCACHE="OFF" \
+    -DAPPEND_CXXFLAGS="-O2" \
```

Comment:
Any reason not to let the CMake defaults handle this?

Reply:
The default *should* do optimization... We should figure out why it isn't, if not!

Reply:
Without any cmake parameters, I see it using `-O2`...